### PR TITLE
[Scrollbar] dont try and arrange the thumb and buttons if the extent …

### PR DESCRIPTION
**Pull request by danwalmsley**
…is zero.

### CODE REVIEW

**Comment by kekekeks** [comment]
Completely skipping the arrange is probably a bad idea, since it might leave old (valid) sizes/positions unchanged. It's better to ensure that all sizes are `>= 0`

File: src/Avalonia.Controls/Primitives/Track.cs [hunk 1]
```diff
@@ -126,19 +126,19 @@ protected override Size ArrangeOverride(Size finalSize)
...
-                var thumbWidth = double.IsNaN(viewportSize) ? thumb?.DesiredSize.Width ?? 0 : finalSize.Width * viewportSize / extent;
-                var remaining = finalSize.Width - thumbWidth;
-                var firstWidth = range <= 0 ? 0 : remaining * offset / range;
-
-                if (decreaseButton != null)
+                if (Orientation == Orientation.Horizontal)
                 {
-                    decreaseButton.Arrange(new Rect(0, 0, firstWidth, finalSize.Height));
-                }
+                    var thumbWidth = double.IsNaN(viewportSize) ? thumb?.DesiredSize.Width ?? 0 : finalSize.Width * viewportSize / extent;
```
Comment (Sorien) [inline src/Avalonia.Controls/Primitives/Track.cs]: is "double.IsNaN(viewportSize)" still needed? same line 154

Comment (danwalmsley) [reply src/Avalonia.Controls/Primitives/Track.cs]: yes definitely needed, when its run through the unit tests, viewport size can be NaN.

latest code simplifies this line though.

**Review by kekekeks — APPROVED** [review]
We probably need to refactor that logic to be in one method to avoid code duplication. Other than that LGTM

**Pull request merged**